### PR TITLE
`CocoaPods`: enabled `visionOS`

### PR DESCRIPTION
**Pull request by NachoSoto**
### Changes:
- Added `visionOS` to list of deployment targets in `Podspec`
- ~~Updated `Swift` version to 5.7 (see https://swiftversion.net), Xcode 14.0 is required, and that's Swift 5.7. We no longer support 5.5~~ (done in a previous PR already)
- Added `visionOS` CocoaPods lint check to `build_visionos`
- Added `VISION_OS` to Podspec (see #2989)
- ~~Changed `visionOS` CI job to use `Xcode 15 beta 8` (see https://discuss.circleci.com/t/xcode-15-rc-released-important-notice-for-visionos-sdk-users/49278)~~ (no longer needed)

Fixes https://community.revenuecat.com/sdks-51/how-to-support-visionos-via-cocoapods-3499

### CODE REVIEW

**Comment by NachoSoto** [comment]
We might have to bump the jobs to Xcode 15.0, but wanted to see whether this is working for now.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1030,6 +1030,10 @@ workflows:
           xcode_version: '14.3.0'
       - build-visionos:
           xcode_version: '15.0.0'
+      - release-checks:
+          xcode_version: '14.3.0'
+      - installation-tests-cocoapods:
+          xcode_version: '14.3.0'
```
Comment (NachoSoto) [inline .circleci/config.yml:1036]: TODO: remove me

File: Tests/InstallationTests/CocoapodsInstallation/Gemfile.lock [hunk 2]
```diff
@@ -32,11 +32,11 @@ GEM
     aws-sigv4 (1.4.0)
       aws-eventstream (~> 1, >= 1.0.2)
     babosa (1.0.4)
-    claide (1.0.3)
-    cocoapods (1.12.1)
+    claide (1.1.0)
+    cocoapods (1.13.0)
```
Comment (NachoSoto) [inline Tests/InstallationTests/CocoapodsInstallation/Gemfile.lock:36]: Needed to update it here too.

**Comment by NachoSoto** [comment]
Blocked by https://github.com/CocoaPods/CocoaPods/pull/11965#issuecomment-1743878949

File: fastlane/Fastfile [hunk 3]
```diff
@@ -238,6 +238,13 @@ platform :ios do
 
   desc "visionOS build"
   lane :build_visionos do |options|
+    pod_lib_lint(
+      verbose: true,
+      podspec:'RevenueCat.podspec', 
+      platforms:"xros",
+      fail_fast: true
+    )
+
```
Comment (aboedo) [inline fastlane/Fastfile:247]: feels like we should also include it here: https://github.com/RevenueCat/purchases-ios/blob/main/fastlane/Fastfile#L937 and here https://github.com/RevenueCat/purchases-ios/blob/main/fastlane/Fastfile#L226

Comment (aboedo) [reply fastlane/Fastfile:247]: unless xcode versions pose an issue

Comment (NachoSoto) [reply fastlane/Fastfile:247]: Yeah I made it separate until we use Xcode 15 for everything. I'll leave a comment.

Comment (NachoSoto) [reply fastlane/Fastfile:247]: Comment added.

File: fastlane/Fastfile [hunk 4]
```diff
@@ -238,6 +238,13 @@ platform :ios do
 
   desc "visionOS build"
   lane :build_visionos do |options|
+    pod_lib_lint(
+      verbose: true,
+      podspec:'RevenueCat.podspec', 
+      platforms:"xros",
```
Comment (NachoSoto) [inline fastlane/Fastfile:244]: Xcode 15.1 beta 3 changes this to `visionos` so this will probably have to change too.

**Comment by NachoSoto** [comment]
Reminder that we need to add this to RevenueCatUI too (see #3368)

File: Tests/InstallationTests/CocoapodsInstallation/Gemfile.lock (hunk 2 above)
Comment (ds3instagram) [reply Tests/InstallationTests/CocoapodsInstallation/Gemfile.lock:36]: Great update! 👏 Excited to see the addition of visionOS to the deployment targets in the Podspec. Thanks for keeping the Swift version up-to-date and providing clear instructions on the required Xcode version. The integration of the visionOS CocoaPods lint check and the switch to Xcode 15 beta 8 for the CI job make it even better.

However, I noticed the mention of the issue CocoaPods/CocoaPods#11965. Any insights on how this is being addressed or any potential workarounds? Looking forward to the continued progress! 🚀

Comment (NachoSoto) [reply Tests/InstallationTests/CocoapodsInstallation/Gemfile.lock:36]: Support for it on CocoaPods is still not working: https://github.com/CocoaPods/CocoaPods/pull/12159#issuecomment-1863576353
We haven't received an update from them, so until then unfortunately we can't release this.

File: .circleci/config.yml [hunk 5]
```diff
@@ -1139,6 +1139,10 @@ workflows:
           <<: *release-branches-and-main
       - build-tv-watch-and-macos:
           xcode_version: '14.3.0'
+      - installation-tests-cocoapods:
```
Comment (NachoSoto) [inline .circleci/config.yml:1142]: TODO: remove

File: RevenueCat.podspec [hunk 6]
```diff
@@ -20,8 +20,13 @@ Pod::Spec.new do |s|
   s.watchos.deployment_target = '6.2'
   s.tvos.deployment_target = '11.0'
   s.osx.deployment_target = '10.13'
+  s.visionos.deployment_target = '1.0'
   
-  s.pod_target_xcconfig = { 'DEFINES_MODULE' => 'YES' }
+  s.pod_target_xcconfig = {
+    'DEFINES_MODULE' => 'YES',
+    'SWIFT_ACTIVE_COMPILATION_CONDITIONS[sdk=xros*]' => '$(inherited) VISION_OS',
```
Comment (NachoSoto) [inline RevenueCat.podspec:27]: This is new.

**Pull request merged**